cmd/prometheus-config-reloader/main.go

```diff
@@ -227,3 +253,91 @@ func createOrdinalEnvvar(fromName string) error {
 	val := reg.FindString(os.Getenv(fromName))
 	return os.Setenv(statefulsetOrdinalEnvvar, val)
 }
+
+type prometheusReloadTracker struct {
+	baseURL *url.URL
+	client  http.Client
+	logger  log.Logger
+
+	lastReload  time.Time
+	lastSuccess bool
+}
+
+func (prt *prometheusReloadTracker) getConfigStatus(ctx context.Context) (bool, time.Time, error) {
+	r, err := http.NewRequestWithContext(ctx, "GET", prt.baseURL.JoinPath("/api/v1/status/runtimeinfo").String(), nil)
+	if err != nil {
+		return false, time.Time{}, err
+	}
+
+	resp, err := prt.client.Do(r)
+	if err != nil {
+		return false, time.Time{}, fmt.Errorf("%s: %w", r.URL.String(), err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode/100 != 2 {
+		return false, time.Time{}, fmt.Errorf("%s: invalid status code: %d", r.URL.String(), resp.StatusCode)
+	}
+
+	var runtimeInfo = struct {
+		Status string `json:"status"`
+		Data   struct {
+			ReloadConfigSuccess bool      `json:"reloadConfigSuccess"`
+			LastConfigTime      time.Time `json:"lastConfigTime"`
+		} `json:"data"`
+	}{}
```

Comment:
Do we know if prometheus has a history of changing the API response here? I wonder if we'll need to have a test to make sure we don't break when upgrading prometheus versions

Reply:
https://prometheus.io/docs/prometheus/latest/stability/

The v1 API should be stable. The API returns these fields since at least late 2019 so I wouldn't check the exact Prometheus version (if it ever happens, the only advice would be to upgrade).